gen/__init__.py

```diff
@@ -306,20 +305,67 @@ def do_gen_package(config, package_filename):
     log.info("Package filename: %s", package_filename)
 
 
-def extract_files_with_path(start_files, paths):
+def render_late_content(content, late_values):
+    # TODO(branden): Unit tests
+
+    def _next_substring(string, substring, start=0):
+        idx = string.find(substring, start)
+        if idx < 0:
+            return None
+        return idx, idx + len(substring)
+
+    def _interpolate_values(string):
+        if not string:
+            return
+
+        # Find the next placeholder.
+        placeholder_start = _next_substring(string, gen.internals.LATE_BIND_PLACEHOLDER_START)
+        if not placeholder_start:
+            # No placeholder found in the string.
+            yield string
+            return
+
+        # Yield the string preceding the placeholder, if any.
+        if placeholder_start[0] > 0:
+            yield string[:placeholder_start[0]]
+
+        # Find the end of the placeholder.
+        placeholder_end = _next_substring(string, gen.internals.LATE_BIND_PLACEHOLDER_END, placeholder_start[1])
+        if not placeholder_end:
+            raise Exception("Bad late config file: Can't find end of placeholder")
```

Comment:
would be good to include the reaminder of the string here so when this error comes up can see where in the string that was used as input the error is.